ifrs17/Report/ReportMutableScopesInteractive.ipynb

```diff
@@ -0,0 +1,580 @@
+                "\n        interactiveObjectCache = new Dictionary<string, Systemorph.Vertex.InteractiveObjects.InteractiveObjectView>();",
+                "\n    }",
+                "\n",
+                "\n    public Systemorph.Vertex.InteractiveObjects.InteractiveObjectView GetFormsEntity<T>(string name = null) where T : ReportScope {",
+                "\n        var key = name ?? typeof(T).Name;",
+                "\n        if(!interactiveObjectCache.TryGetValue($\"{key}FormsEntity\", out var ret))",
+                "\n            ret = interactiveObjectCache[$\"{key}FormsEntity\"] = interactiveObject.CreateView($\"{key}FormsEntity\", state => state.GetScope<T>(key, o => o.WithStorage(storage)));",
+                "\n        return ret;",
+                "\n    }",
+                "\n",
+                "\n    // public Systemorph.Vertex.InteractiveObjects.InteractiveObjectView GetReport<T>(string name = null) where T : ReportScope {",
+                "\n    //     var key = name ?? typeof(T).Name;",
+                "\n    //     if(!interactiveObjectCache.TryGetValue(key, out var ret))",
+                "\n    //     ret = interactiveObjectCache[key] = interactiveObject.CreateView(key,",
+                "\n    //         state => {",
+                "\n    //             var scope = state.GetScope<T>(key, o => o.WithStorage(storage));",
+                "\n    //             //var filters = scope.GetFilters(); // Not used and should be improved",
+                "\n    //             return scope.ToReportAsync();",
+                "\n    //         });",
+                "\n    //     return ret;",
+                "\n    // }",
+                "\n",
+                "\n    public ReportScope GetReportScope<T>() where T : ReportScope => interactiveObject.State.GetScope<T>(typeof(T).Name, o => o.WithStorage(storage));",
+                "\n}"
+            ],
+            "metadata": {},
+            "execution_count": 0,
+            "outputs": []
+        },
+        {
+            "cell_type": "code",
+            "source": [
+                "// public class Ifrs17Interactive ",
+                "\n// {",
+                "\n//     //private Systemorph.Vertex.Scopes.Proxy.IScopeFactory scopes;",
+                "\n//     private Systemorph.Vertex.Pivot.Builder.Interfaces.IPivotFactory report;",
+                "\n//     private Systemorph.InteractiveObjects.InteractiveObjectVariable interactiveObject;",
+                "\n//     private ReportStorage storage;",
+                "\n    ",
+                "\n//     //reset",
+                "\n//     public void Reset(IWorkspace workspace) => storage = new ReportStorage(workspace, report);",
+                "\n",
+                "\n//     //constructor",
+                "\n//     public Ifrs17Interactive (IWorkspace workspace, Systemorph.Vertex.Pivot.Builder.Interfaces.IPivotFactory report, Systemorph.InteractiveObjects.InteractiveObjectVariable interactiveObject)",
```

Comment:
This constructor looks someone cumbersome to me. We take the Workspace, Report and IntreactiveObject from the platform. Can we define all three these objects and static and public and define them immediately after the class is defines (before initialization). The we could have a parameterless constructor here.

Reply:
No, this won't work as we need their instances. With the static approach, you cannot control what data is in. It is the same concept we have in all storages.